File: String/MostFrequentCharacters/mostFrequentCharacters.cpp
#include <algorithm>
#include <string>
using namespace std;

class Solution {
  public:
    char getMaxOccuringChar(string& s) {
        //  code here
        sort(s.begin(),s.end());
        int max = 0;
        int count = 1;
        char ans = '\0';
        char currChar = s[0];
        for(int i=1;i<s.length();i++){
            if(s[i]!=currChar){
                if(count>max){
                    ans = currChar;
                    max = count;
                }
                count = 1;
                currChar = s[i];
            }
            else count++;
        }
        if(count>max){
            ans = s[s.length()-1];
        }
        return ans;
    }
};

File: String/MostFrequentCharacters/mostFrequentCharacters.py
class Solution:
    def getMaxOccurringChar(self, s):
        #code here
        t = "".join(sorted(s))
        count = 1
        maxCount = 1
        ans = t[0]
        curr = t[0]
        for i in range(1,len(s)):
            if curr != t[i]:
                if count > maxCount:
                    maxCount = count
                    ans = curr
                count = 1
                curr = t[i]
            else: count += 1
        if count>maxCount:
            ans = curr
        return ans